native/cocos/scene/ReflectionProbe.cpp

```diff
@@ -184,8 +208,90 @@ void ReflectionProbe::destroy() {
         _realtimePlanarTexture->destroy();
         _realtimePlanarTexture = nullptr;
     }
+    for (const auto& rt : _bakedCubeTextures) {
+        rt->destroy();
+    }
+    _bakedCubeTextures.clear();
+
     scene::ReflectionProbeManager::getInstance()->unRegisterProbe(this);
 }
 
+void ReflectionProbe::initBakedTextures() {
+    if (_bakedCubeTextures.empty()) {
+        for (uint32_t i = 0; i < 6; i++) {
+            auto* rt = ccnew RenderTexture();
+            IRenderTextureCreateInfo info;
+            info.name = "capture";
+            info.height = _resolution;
+            info.width = _resolution;
+            rt->initialize(info);
+            _bakedCubeTextures.emplace_back(rt);
+        }
+    }
+}
+void ReflectionProbe::resetCameraParams() {
+    _camera->setProjectionType(CameraProjection::PERSPECTIVE);
+    _camera->setOrthoHeight(10.F);
+    _camera->setNearClip(1.F);
+    _camera->setFarClip(1000.F);
+    _camera->setFov(static_cast<float>(mathutils::toRadian(90.F)));
+    _camera->setPriority(0);
+    if (!_bakedCubeTextures.empty()) {
+        _camera->changeTargetWindow(_bakedCubeTextures[0]->getWindow());
+    }
+    _camera->resize(_resolution, _resolution);
+    _camera->setVisibility(_visibility);
+
+    _camera->setClearColor(_backgroundColor);
+    _camera->setClearDepth(1.F);
+    _camera->setClearStencil(0.F);
+    _camera->setClearFlag(_clearFlag);
+
+    _camera->setAperture(CameraAperture::F16_0);
+    _camera->setShutter(CameraShutter::D125);
+    _camera->setIso(CameraISO::ISO100);
+
+    _cameraNode->setWorldPosition(_node->getWorldPosition());
+    _cameraNode->setWorldRotation(_node->getWorldRotation());
+    _camera->update(true);
+}
+void ReflectionProbe::captureCubemap() {
+    initBakedTextures();
+    resetCameraParams();
+    packBackgroundColor();
+    _needRender = true;
+}
+
+void ReflectionProbe::updateCameraDir(int32_t faceIdx) {
+    _cameraNode->setRotationFromEuler(CAMERA_DIR[faceIdx]);
+    _camera->update(true);
+}
+
+Vec2 ReflectionProbe::renderArea() const {
+    if (_probeType == ProbeType::PLANAR) {
+        return Vec2(_realtimePlanarTexture->getWidth(), _realtimePlanarTexture->getHeight());
+    }
+    return Vec2(_resolution, _resolution);
+}
+
+void ReflectionProbe::packBackgroundColor() {
+    Vec3 rgb = Vec3(_camera->getClearColor().x, _camera->getClearColor().y, _camera->getClearColor().z);
```

Comment:
Can just initialize it like
```c++
Vec3 rgb{_camera->getClearColor().x, _camera->getClearColor().y, _camera->getClearColor().z};
```
to avoid creating a temple variable and doing assignment operation.